src/modules/tools/zprobe/ZProbe.cpp

```diff
@@ -287,17 +287,35 @@ void ZProbe::on_gcode_received(void *argument)
 
     } else if(gcode->has_m) {
         // M code processing here
-        if(gcode->m == 119) {
-            int c = this->pin.get();
-            gcode->stream->printf(" Probe: %d", c);
-            gcode->add_nl = true;
-
-        }else {
-            for(auto s : strategies){
-                if(s->handleGcode(gcode)) {
-                    return;
+        int c;
+        switch (gcode->m) {
+            case 119:
+                c = this->pin.get();
+                gcode->stream->printf(" Probe: %d", c);
+                gcode->add_nl = true;
+                break;
+
+            case 670:
+                if (gcode->has_letter('S')) this->slow_feedrate = gcode->get_value('S');
+                if (gcode->has_letter('F')) this->fast_feedrate = gcode->get_value('F');
```

Comment:
cannot use F here as it is globally reserved to mean G0/G1 feedrate which can affect other things.